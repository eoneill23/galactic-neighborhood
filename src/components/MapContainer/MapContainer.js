import React, { Component } from 'react';
import { googleKey } from '../../keys';
import { Map, GoogleApiWrapper, Marker } from 'google-maps-react';
import PropTypes from 'prop-types';

const mapStyles = {
  height: '270px',
  width: '300px'
};

export class MapContainer extends Component {
  
  render() {
    return (
      <Map
        google={this.props.google}
        zoom={3}
        style={mapStyles}
        initialCenter={{ lat: this.props.lat, lng: this.props.long }}
      >
        <Marker position={{ lat: this.props.lat, lng: this.props.long }} />
      </Map>
    )
  }
}

export default GoogleApiWrapper({
  apiKey: googleKey
})(MapContainer);

MapContainer.propTypes = {
  google: PropTypes.object,
  lat: PropTypes.number,
  long: PropTypes.number
}