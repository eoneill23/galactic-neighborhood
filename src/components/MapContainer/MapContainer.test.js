import React from 'react';
import { shallow } from 'enzyme';
import { MapContainer } from './MapContainer';

describe('MapContainer', () => {

  it('should match the snapshot with all data passed in correctly', () => {

    let wrapper = shallow(<MapContainer google={{maps: {}}} lat={1} long={2}/>);

    expect(wrapper).toMatchSnapshot();
  });
});