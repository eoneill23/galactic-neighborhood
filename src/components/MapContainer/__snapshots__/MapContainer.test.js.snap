// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer should match the snapshot with all data passed in correctly 1`] = `
<Map
  center={Object {}}
  centerAroundCurrentLocation={false}
  containerStyle={Object {}}
  google={
    Object {
      "maps": Object {},
    }
  }
  initialCenter={
    Object {
      "lat": 1,
      "lng": 2,
    }
  }
  style={
    Object {
      "height": "270px",
      "width": "300px",
    }
  }
  visible={true}
  zoom={3}
>
  <Marker
    name="Marker"
    position={
      Object {
        "lat": 1,
        "lng": 2,
      }
    }
  />
</Map>
`;
